File: Python_Programs/guessthenumber.py
def replay():
    try:

        print("Welcome to the number guessing game.")
        print("You have 5 attemps to guess the correct number.")
        n = 18
        attempts = 1
        while attempts <= 5:
            num = int(input("Enter the correct number:\n"))
            if num < n:
                print("You entered a smaller number. Try again!")
            elif num > n:
                print("You entered a greater number. Try again!")
            else:
                print("You won with",5-attempts, "attempts left and using",attempts, "attempts!")
                break
            attempts = attempts + 1


        if attempts > 5:
            print("You lost!")


        rerun = input("Do you want to play again?('y' or 'n')")
        if rerun == 'y':
            print("Thanks for playing again!")
            replay()
        elif rerun == 'n':
            print("Exited Successfully!")
            exit()
        else:
            print("Something went wrong!")

    except Exception as error1:
        print("Something went wrong!")
replay()

File: Java_Programs/guess_the_number.java
package com.company;

import java.util.Random;
import java.util.Scanner;

public class guess_the_number {
    public static void main(String[] args) {
        Random ran = new Random();
        Scanner sc = new Scanner(System.in);
        try {
            int random_no = ran.nextInt(101);
            System.out.println("You have ten attempts to guess the correct number less then 100!");
            int attempts = 0;
            while (attempts <= 9) {
                System.out.println("Enter a number less then 100!");
                int user_choice = sc.nextInt();
                if (user_choice > 100) {
                    System.out.printf("%d is greater than 100\n", user_choice);
                }
                else if (user_choice < random_no){
                    System.out.println("You entered a smaller number!");
                }
                else if (user_choice > random_no){
                    System.out.println("You entered a greater number!"); 
                }
                else if (user_choice == random_no){
                    System.out.println("You won!");
                    break;
                }
                attempts++;
            }
            if (attempts > 9){
                System.out.println("You lost!");
                System.out.printf("The number was %d", random_no);
            }
        } catch (Exception error1) {
            System.out.println("Again invalid input");
        }
    }
}
